src/gensxg.c

```diff
@@ -483,9 +493,9 @@ void write_sxg(const sxg_signer_list_t* signers,
     }
   }
 
-  size_t written = fwrite(result.data, sizeof(uint8_t), result.size, out);
-  if (written != result.size) {
-    perror("fwrite");
+  write_buffer(&result, out);
```

Comment:
You don't have to factor out write_buffer now as out/stdout path are combined. It's your call to keep factoring out or embed again.

Reply:
OK, back to inlined.